File: TS/nes_exploration/handlers.ts
import { EventLogger } from "./eventLogger";

const logger = new EventLogger();

export function handleLogin(username: string): void {
  /** Log a successful login event. */
  logger.logEvent("auth", `${username} logged in`);
}

export function handleError(errorCode: number, details: string): void {
  /** Log an error event. */
  logger.logEvent("system", `Error ${errorCode}: ${details}`);
}

export function handleRequest(endpoint: string, method: string): void {
  /** Log an incoming API request. */
  logger.logEvent("api", `${method} ${endpoint}`);
}

File: Python/nes_exploration/handlers.py
from event_logger import EventLogger


logger = EventLogger()


def handle_login(username: str):
    """Log a successful login event."""
    logger.log_event("auth", f"{username} logged in")


def handle_error(error_code: int, details: str):
    """Log an error event."""
    logger.log_event("system", f"Error {error_code}: {details}")


def handle_request(endpoint: str, method: str):
    """Log an incoming API request."""
    logger.log_event("api", f"{method} {endpoint}")
